curriculum/challenges/english/04-data-visualization/json-apis-and-ajax/render-images-from-data-sources.english.md

```diff
@@ -104,8 +105,65 @@ tests:
 ## Solution
 <section id='solution'>
 
-```js
-// solution required
+```html
+<script>
+  document.addEventListener('DOMContentLoaded', function(){
+    document.getElementById('getMessage').onclick = function(){
+      const req = new XMLHttpRequest();
+      req.open("GET",'/json/cats.json',true);
+      req.send();
+      req.onload = function(){
+        const json = JSON.parse(req.responseText);
+        let html = "";
+        json.forEach(function(val) {
+          html += "<div class = 'cat'>";
+          // Add your code below this line
+html += "<img src = '" + val.imageLink + "' " + "alt='" + val.altText + "'>";
+
```

Comment:
This could probably be indented as well.

Reply:
Resolved in a separate commit.